Zend/tests/match/014.phpt

```diff
@@ -0,0 +1,16 @@
+--TEST--
+Error when using result of match statement list
+--FILE--
+<?php
+
+var_dump(match ('bar') {
+    'foo' => 'foo',
+    'bar' => 'bar',
+    'baz' => {
+        var_dump('This should not compile');
+    },
+});
+
+?>
+--EXPECTF--
+Fatal error: Match expressions that are not statements can't contain statement lists in %s on line %d
```

Comment:
I was confused about what exactly changed yesterday and how from the vote email - posting this in case others were confused by this.

Yesterday, this was added to https://wiki.php.net/rfc/match_expression#blocks

> **For the time being using blocks in match expressions that use the return value in any way results in a compilation error:**

https://externals.io/message/109842 (`[VOTE] match expression`)

> Thus I have decided to move block expressions to a separate RFC. 

Also, I was a bit confused about the email and whether "block expressions" referred to return values from blocks. **It actually meant the use of blocks in match expressions(?)**, as described in the RFC.

Reply:
Sorry. "block expressions" refers to blocks with return values as match arms. That distinction wasn't clear enough in the e-mail.